System/Random.hs

```diff
@@ -30,23 +14,15 @@ module System.Random
   -- * Introduction
   -- $introduction
 
-  -- * Usage
-  -- ** How to generate pseudo-random values in monadic code
-  -- $usagemonadic
-
-  -- ** How to generate pseudo-random values in pure code
-  -- $usagepure
-
-  -- * Pure and monadic pseudo-random number generator interfaces
+  -- * Pure number generator interface
   -- $interfaces
     RandomGen(..)
-  , MonadRandom(..)
-  , Frozen(..)
-  , runGenM
-  , runGenM_
-  , RandomGenM(..)
-  , splitRandomGenM
-
+  , uniform
+  , uniformR
+  , genByteString
+  , Random(..)
+  , Uniform
```

Comment:
In #96, I decided not to export `Uniform` and `UniformRange` from `System.Random` because their definition uses `MonadRandom`:

```haskell
class Uniform a where
  uniformM :: MonadRandom g s m => g s -> m a
```

If a user only imports `System.Random`, it is not clear to me how they could use `uniformM` given that `MonadRandom` will not be in scope. Am I missing something here?

Same applies to `UniformRange`.

Reply:
There are two functions exported couple lines above `uniform` and `uniformR`. They allow usage of instances of `Uniform` and `UniformRange` type classes with pure `RandomGen` while hiding the fact that it is done through the `MonadRandom`. Which is pretty neat I think ;)

But because definitions `uniformM` and `uniformRangeM` use  `MonadRandom` for implementation, they are not exported from `System.Random`, so you are partially right.